### DISCUSSION TITLE
Kobo: Sending books to hidden directory via Calibre?

### DISCUSSION THREAD

**Discussion by jonnyl2** [koreader/koreader#12532] [Q&A] [answered]
Since many of you use Kobos and Calibre, and it is ~~recommended~~ possible to put books into a hidden directory to hide them from Nickel, how do I tell Calibre to put the books in said hidden directory?

If I specify `.Books/`, it changes it to `_Books/`. And if I put all in quotes, `'.Books/'`, then it just becomes `_.Books/`. 

There surely must be a way?

**Comment by Frenzie** [comment]
> and it is recommended to put books into a hidden directory to hide them from Nickel

Is it? It seems more of a self-evident choice depending on what you want. ;-)

But that aside, I don't know about calibre. Is it perhaps doing something weird because the directory already exists?

**Comment by jonnyl2** [comment]
>Is it? It seems more of a self-evident choice depending on what you want. ;-)

True, but I first learned about this via KOReader's documentation :-)

>Is it perhaps doing something weird because the directory already exists?

Unfortunately no; I just tested with a different, non-existing directory name and it's doing the same thing. :-/

**Reply by Frenzie** [reply]
What does it do with `.Books/subfolder`?

**Reply by jonnyl2** [reply]
It doesn't make a difference if I tack on additional subfolders, unfortunately.

**Comment by NiLuJe** [comment]
> and it is recommended to put books into a hidden directory

I *strongly* recommend people *not* attempt to do that, actually.

**Reply by jonnyl2** [reply]
Really, why is that? Nickel updating the library on each restart is a bit annoying if you never use it.

**Reply by NiLuJe** [reply]
Nickel barely ever refreshes the library on its own, and it certainly never does so on startup.

The only thing that will refresh the library is a (Nickel) USBMS session (or an explicit refresh via NickelMenu).

I was never a fan of the "hide your Library from Nickel" thing to begin with, but it became even more of a footgun after FW >= 4.17, so, meh.
Whatever comments you might find about that are probably very very very old.

There's no real reason do to that anymore, as you don't really gain anything from it, and unless you *really* know what you're doing and understand the implications, I strongly recommend people just leave things as they are and just enjoy reading instead ;).

**Reply by jonnyl2** [reply]
What changed after FW 4.17?

Edit: [Found it](https://github.com/koreader/koreader/issues/5430)

**Reply by Frenzie** [reply]
> I was never a fan of the "hide your Library from Nickel" thing to begin with,

It's not for the fun of it. Being locked out for many minutes is an atrocious experience. 

> There's no real reason do to that anymore

So I assume you mean indexing performance is at least some 1000 x better, and that's not a typo or an exaggeration.

**Reply by NiLuJe** [reply]
There's *no* auto indexing, so I fail to see reasoning for even wanting to do that in the first place :?

**Reply by jonnyl2** [reply]
But there is, at least after using USBMS under Nickel? If you're like me and have nearly a dozen patches and tweaks, you're bound to run into breaking KOReader updates that have to be fixed from the Nickel USB interface.

**Reply by NiLuJe** [reply]
I mean, to each their own, but given that SSH is a thing, I pretty much never ever run an USBMS session under Nickel unless I want to.

I *also* don't see what the fuss about hiding stuff from Nickel is to begin with, but clearly I must be doing things differently than either of you ;).

**Reply by NiLuJe** [reply]
The point I'm trying to make is that I don't *recommend* people do that *by default* (because it's needless complication, and can actually turn into a massive footgun if Nickel has a sad).

If, somehow, it helps your actual workflow, sure, go ahead ;p.

**Reply by Frenzie** [reply]
> The point I'm trying to make is that I don't recommend people do that by default.

I think you just did the exact opposite tbh, lol

**Reply by Frenzie** [reply]
So you need to either extremely carefully perform magical combinations of incantations the way @NiLuJe does to avoid triggering indexing (=impossible) or you can just hide it.

**Reply by jonnyl2** [reply]
Point taken. I edited the OP so as not to give the impression that this is KOReader's official stance ;)

**Comment by NiLuJe** [comment]
So, to get back to your actual query:

> If I specify `.Books/`

Specify *where*?

Also, what are we talking about here? Send To Device from calibre via USB? Via the Wireless client? Via OPDS?

**Reply by jonnyl2** [reply]
In the device settings on the Linux GUI:

![image](https://github.com/user-attachments/assets/fcbe1f44-6719-4de2-be81-e7329049a18a)

I was using USB to transfer the books, but this setting also applies to other transfer methods AFAIK.

**Reply by NiLuJe** [reply, marked as answer]
Welp, apparently, yeah, by design:

https://github.com/kovidgoyal/calibre/blob/00ddbdb15970a48ffbaa76f8a135bd6f5ac919f5/src/calibre/__init__.py#L154-L156

Short of mangling a driver to set a custom root path from there (specifically, `EBOOK_DIR_MAIN`), I don't have any bright ideas.

Unless someone who actually did something of the sort chimes in ;).

**Reply by Frenzie** [reply]
I can kinda see where that's coming from I suppose, but in Windows you need to go through special effort to make a file that starts with a period.

**Reply by jonnyl2** [reply]
Thank you for getting to the root of this. I marked it as answered but of course I'd be happy if anyone has any solutions.

Or is it perhaps possible to change the Kobo config file `.kobo/Kobo/Kobo eReader.conf` setting 

`ExcludeSyncFolders=(\\.(?!kobo|adobe).+|([^.][^/]*/)+\\..+)` 

to also exclude a folder named `calibre/`? I don't want to mess with that regex without knowing exactly what I'm doing.

**Reply by NiLuJe** [reply]
That is definitely an option if you can manage not to bork the regex.

What may also be an option is switch to the standard USB folder (insert its actual name here ;p) driver instead of the Kobo driver and point that at your hidden folder.

**Reply by jonnyl2** [reply]
Sorry, I don't understand what you mean by that second part.

**Reply by NiLuJe** [reply]
I meant something along those lines: https://github.com/koreader/koreader/issues/12062#issuecomment-2190044070

**Reply by jonnyl2** [reply]
That's the perfect solution, thank you! I didn't know Calibre could treat a folder like a device.

**Reply by pazos** [reply]
> That's the perfect solution, thank you! I didn't know Calibre could treat a folder like a device.

That's indeed the perfect solution. It also makes possible to share the same library wired and wirelessly (but make sure first you use the same template on both, if you want to keep things organized ;))

**Reply by NiLuJe** [reply]
Unless you *need* the Nickel db for... reasons ;o).
